gc/base/standard/ConcurrentGC.cpp

```diff
@@ -174,19 +174,21 @@ con_helper_thread_proc2(OMRPortLibrary* portLib, void *info)
 } /* extern "C" */
 
 void
-MM_ConcurrentGC::reportGCCycleStart(MM_EnvironmentStandard *env)
+MM_ConcurrentGC::reportGCCycleStart(MM_EnvironmentBase *env)
 {
 	/* TODO CRGTMP - Temporarily move this code to the point when
 	 * we have exclusiveVMAccess so that GC check works.  Move it back
 	 * once we complete JAZZ 33760.
 	 */
+	MM_CycleState *previousCycleState = env->_cycleState;
+
 	_concurrentCycleState = MM_CycleState();
 	env->_cycleState = &_concurrentCycleState;
 	env->_cycleState->_type = _cycleType;
 
 	MM_ParallelGlobalGC::reportGCCycleStart(env);
 
-	env->_cycleState = NULL;
+	env->_cycleState = previousCycleState;
```

Comment:
This feels like a fix that we do not understand.  I would like to understand how this worked in Java so that we can understand if this is the right fix